Framework/DataHandling/test/LoadSampleShapeTest.h

```diff
@@ -0,0 +1,174 @@
+#ifndef LOAD_SHAPETEST_H_
+#define LOAD_SHAPETEST_H_
+
+#include "MantidAPI/FrameworkManager.h"
+#include "MantidAPI/AnalysisDataService.h"
+#include "MantidAPI/Sample.h"
+#include "MantidDataHandling/LoadSampleShape.h"
+#include "MantidDataHandling/LoadInstrument.h"
+#include "MantidGeometry/Objects/MeshObject.h"
+#include "MantidKernel/OptionalBool.h"
+#include "MantidTestHelpers/ComponentCreationHelper.h"
+#include "MantidTestHelpers/WorkspaceCreationHelper.h"
+#include <cxxtest/TestSuite.h>
+
+using namespace Mantid;
+using namespace Mantid::API;
+using namespace Mantid::DataHandling;
+using namespace Mantid::DataObjects;
+using namespace Mantid::Geometry;
+
+class LoadSampleShapeTest : public CxxTest::TestSuite {
+public:
+  static LoadSampleShapeTest *createSuite() {
+    return new LoadSampleShapeTest();
+  }
+  static void destroySuite(LoadSampleShapeTest *suite) { delete suite; }
+
+  void testName() { TS_ASSERT_EQUALS(loadShape.name(), "LoadSampleShape"); }
+
+  void testVersion() { TS_ASSERT_EQUALS(loadShape.version(), 1); }
+
+  void testInit() {
+
+    TS_ASSERT_THROWS_NOTHING(loadShape.initialize());
+    TS_ASSERT(loadShape.isInitialized());
+
+    TSM_ASSERT_EQUALS("should be 3 properties here", 3,
+                      (size_t)(loadShape.getProperties().size()));
+  }
+
+  void testConfidence() {
+    LoadSampleShape testLoadSampleShape;
+    testLoadSampleShape.initialize();
+    testLoadSampleShape.setPropertyValue("Filename", "cube.stl");
+    std::string path = testLoadSampleShape.getPropertyValue("Filename");
+    auto *descriptor = new Kernel::FileDescriptor(path);
+    TS_ASSERT_EQUALS(90, testLoadSampleShape.confidence(*descriptor));
+    delete descriptor;
+  }
+
+  void testExec_2WS() {
+    LoadSampleShape testLoadSampleShape;
+    testLoadSampleShape.initialize();
+    testLoadSampleShape.setPropertyValue("Filename", "cube.stl");
+    prepareWorkspaces(testLoadSampleShape, false);
+    TS_ASSERT_THROWS_NOTHING(testLoadSampleShape.execute());
+    TS_ASSERT(testLoadSampleShape.isExecuted());
```

Comment:
This test only checks that the algorithm did not throw, this is almost pointless.